scripts/test-emoji-validation.sh

```
@@ -0,0 +1,197 @@
+    "add unit tests for user authentication"
+    "update build configuration for production"
+    "optimize GitHub Actions workflow"
+    "clean up unused dependencies"
+    "improve code formatting and style"
+    "revert previous breaking changes"
+    "version 1.0.0 with new features"
+)
+
+# Current regex patterns from commit-msg hook
+COMMIT_PATTERN_WITH_EMOJI='^[^[:space:]]+[[:space:]][a-z]+:[[:space:]].+'
+COMMIT_PATTERN_WITHOUT_EMOJI='^[a-z]+: .+'
```

Shellcheck (0.10.0)

[warning] COMMIT_PATTERN_WITHOUT_EMOJI appears unused. Verify use (or export if used externally).

(SC2034)